analyzer/src/main/kotlin/managers/Pub.kt

```diff
@@ -390,11 +391,19 @@ class Pub(
         if (!definitionFile.isFile) return emptyList()
 
         return analyzerResultCacheAndroid.getOrPut(packageName) {
+            val packageManagers = analyzerConfig.packageManagers?.toMutableMap() ?: mutableMapOf()
+            val gradleConfiguration = packageManagers["Gradle"] ?: PackageManagerConfiguration()
+            val gradleOptions = gradleConfiguration.options?.toMutableMap() ?: mutableMapOf()
+            val gradleVersion = gradleOptions.getOrPut("gradleVersion") { GRADLE_VERSION }
+
             logger.info {
-                "Analyzing Android dependencies for package '$packageName' using Gradle version $GRADLE_VERSION."
+                "Analyzing Android dependencies for package '$packageName' using Gradle version $gradleVersion."
             }
 
-            Gradle("Gradle", androidDir, analyzerConfig, repoConfig, GRADLE_VERSION)
+            packageManagers["Gradle"] = gradleConfiguration.copy(options = gradleOptions)
+            val gradleAnalyzerConfig = analyzerConfig.copy(packageManagers = packageManagers)
```

Comment:
Please extract the new logic to a function `fun AnalyzerConfiguration.setGradleVersion(String): AnalyzerConfiguration` for readability of this large `return` block.

Reply:
I chose the more generic name `withPackageManagerOption`.